### PR TITLE
Rc1 Fixes

### PR DESCRIPTION
**Pull request by pvelesko**
* Incorporated all outstanding PRs except for graphs
* Various other fixes

### CODE REVIEW

File: src/CHIPBindings.cc [hunk 1]
```diff
@@ -3700,25 +3719,26 @@ hipError_t hipMemcpy3DAsyncInternal(const struct hipMemcpy3DParms *Params,
...
+  if (YSize * XSize == 0)
+    return hipSuccess;
+
+  size_t srcSlicePitch = SrcPitch * YSize;
+  size_t dstSlicePitch = DstPitch * YSize;
+  if ((WidthInBytes == DstPitch) && (WidthInBytes == SrcPitch))
+    ChipQueue->memCopyAsync(DstPtr, SrcPtr, WidthInBytes * Height * Depth);
+  else {
+    // TODO What am I doing wrong here?
+    // ChipQueue->memCopy3DAsync(DstPtr, DstPitch, dstSlicePitch, SrcPtr, SrcPitch, srcSlicePitch, Width, Height, Depth);
```
Comment (pvelesko) [inline src/CHIPBindings.cc:3732]: @pjaaskel  please take a look

File: src/CHIPBindings.cc [hunk 2]
```diff
@@ -3622,6 +3632,8 @@ hipError_t hipMemcpyHtoA(hipArray *DstArray, size_t DstOffset,
 hipError_t hipMemcpy3DAsyncInternal(const struct hipMemcpy3DParms *Params,
                                     hipStream_t Stream) {
   auto ChipQueue = Backend->findQueue(static_cast<chipstar::Queue *>(Stream));
+  LOCK(ChipQueue->QueueMtx); // prevent interruptions
+
   if (ChipQueue->captureIntoGraph<CHIPGraphNodeMemcpy>(Params)) {
     return hipSuccess;
```
Comment (pjaaskel) [inline src/CHIPBindings.cc:3638]: UNLOCK for QueueMtx?? Perhaps lock guard?

Comment (pvelesko) [reply src/CHIPBindings.cc:3638]: LOCK is a macro which creates a lock guard

File: src/CHIPBindings.cc (hunk 1 above)
Comment (pjaaskel) [reply src/CHIPBindings.cc]: @franz you have better grasp of this API, perhaps an educated guess from you here?

**Pull request merged**